lib/rucio/common/cache.py

```diff
@@ -65,3 +68,66 @@ def make_region_memcached(expiration_time, function_key_generator=None):
         region.configure('dogpile.cache.null')
 
     return region
+
+
+def _ignore_arguments_key_generator(ignore_arguments: List[str], namespace, fn: Callable, to_str: Callable[..., str] = str) -> Callable[..., str]:
+    """
+    This function is used to construct a
+    :py:func:`dogpile.cache.FunctionKeyGenerator` instance. This instance can be
+    used to cache function results based on some, possibly not all, keys.
+
+    :param ignore_arguments: The list of arguments to ignore, keyword arguments only.
+    :param namespace: The list of arguments to ignore, keyword arguments only.
+    :param fn: The list of arguments to ignore, keyword arguments only.
+    :param to_str: The function to use to generate a string from the provided value.
+    :returns: A :py:func:`dogpile.cache.FunctionKeyGenerator` implementation.
+    """
+
+    dogpile_cache_key_generator = dogpile_util.kwarg_function_key_generator(namespace, fn, to_str=to_str)
+
+    @wraps(fn)
+    def generate_key(*args, **kwargs):
+        """
+        Construct a cache-key wihtout the specified keyword arguments in
+        `ignore_arguments`.
+
+        Only keyword-arguments can be excluded, since `arg` does not specify the
+        argument name. This is not a problem, since positional arguments should
+        be in the cache key, to avoid big unstructured functions.
+        """
+        filtered_kwargs = {k: kwargs[k] for k in kwargs if k not in ignore_arguments}
+        return dogpile_cache_key_generator(*args, **filtered_kwargs)
+
+    return generate_key
+
+
+def ignore_arguments_key_generator(ignore_arguments: List[str]) -> Callable[[str, Callable], str]:
+    """
+    Constructs a :py:func:`dogpile.cache.FunctionKeyGenerator` function to
+    generate cache-keys which ignores the specified keyword arguments.
+
+    This function can be used together with
+    :py:func:`dogpile.cache.region.CacheRegion.cache_on_arguments` to cache
+    function results based on some (not all) arguments::
+
+        @REGION.cache_on_arguments(function_key_generator=ignore_arguments_key_generator(["session"]))
+        @session
+        def get_rse_attribute(rse_id, key, *, session=None):
+            ...
```

Comment:
I feel like this is too bothersome to write. In this case I'd be in favor of another wrapper to simplify usage:

```python3
def cache(*, ignore: str | Iterable[str] = ''):
    ignored_arg_names: tuple[str, ...]
    if isinstance(ignore, str):
        ignored_arg_names = tuple(ignore.split(','))
    else:
        ignored_arg_names = tuple(ignore)
    return partial(
        REGION.cache_on_arguments, 
        function_key_generator=ignore_arguments_key_generator(ignored_arg_names))
```

Which can then be used as 

```python3
@cache(ignore='session')
def get_rse_attribute(..., session = None):
    ...
```

(note not tested, hopefully got number of partials and wrapper layers right)